springdoc-openapi-starter-common/src/main/java/org/springdoc/ui/AbstractSwaggerWelcome.java

```diff
@@ -134,12 +136,17 @@ else if (swaggerUiConfigParameters.isValidUrl(swaggerUiUrl))
 			else
 				swaggerUiConfigParameters.addUrl(apiDocsUrl);
 			if (!CollectionUtils.isEmpty(swaggerUiConfig.getUrls())) {
-				swaggerUiConfig.cloneUrls().forEach(swaggerUrl -> {
-					swaggerUiConfigParameters.getUrls().remove(swaggerUrl);
-					if (!swaggerUiConfigParameters.isValidUrl(swaggerUrl.getUrl()))
-						swaggerUrl.setUrl(buildUrlWithContextPath(swaggerUrl.getUrl()));
-					swaggerUiConfigParameters.getUrls().add(swaggerUrl);
-				});
+				swaggerUiConfig.cloneUrls()
+						.stream()
+						.filter(swaggerUrl -> !swaggerUiConfigParameters.isValidUrl(swaggerUrl.getUrl()))
+						.forEach(swaggerUrl -> {
+							final var url = buildUrlWithContextPath(swaggerUrl.getUrl());
+							if (!Objects.equals(url, swaggerUrl.getUrl())) {
+								swaggerUiConfigParameters.getUrls().remove(swaggerUrl);
```

Comment:
In some scenarios, I believe this will still encounter the reported ArrayIndexOutOfBoundsException in https://github.com/springdoc/springdoc-openapi/issues/2509. I can duplicate that in the test file by adding `@TestPropertySource(properties = {"spring.mvc.servlet.path=test"})` to the class and you'll hit both ArrayIndexOutOfBoundsExceptions and ConcurrentModificationExceptions depending on the thread.

Would it make more sense to do 
`swaggerUiConfigParameters.getUrls().stream().filter(o -> o.equals(swaggerUrl)).forEach(subUrl -> subUrl.setUrl(url));`
instead?

Reply:
Good point, you are right, I missed it. It would be safer to avoid mutation of the collection at all, instead only the url itself should be mutated when necessary. Thank you! Fixed in https://github.com/springdoc/springdoc-openapi/pull/2668/commits/570e367b4ad4b51cb2efc80815b3a97aad8cba74.